arangosh/Backup/BackupFeature.cpp

```diff
@@ -0,0 +1,509 @@
+  using arangodb::options::VectorParameter;
+
+  options->addOption("--operation",
+                     "operation to perform (may be specified as positional "
+                     "argument without '--operation')",
+                     new DiscreteValuesParameter<StringParameter>(&_options.operation, ::Operations),
+                     static_cast<std::underlying_type<Flags>::type>(Flags::Hidden));
+
+  options->addOption("--force",
+                     "whether to attempt to continue in face of errors (may "
+                     "result in inconsistent backup state)",
+                     new BooleanParameter(&_options.force));
+
+  options->addOption("--identifier", "a unique identifier for a backup",
+                     new StringParameter(&_options.identifier));
+
+  //  options->addOption("--include-search", "whether to include ArangoSearch data",
+  //                     new BooleanParameter(&_options.includeSearch));
+
+  options->addOption(
+      "--label",
+      "an additional label to add to the backup identifier upon creation",
+      new StringParameter(&_options.label));
+
+  options->addOption("--max-wait-for-lock",
+                     "maximum time to wait (in seconds) to aquire a lock on "
+                     "all necessary resources",
+                     new DoubleParameter(&_options.maxWaitForLock));
+
+  options->addOption(
+      "--max-wait-for-restart",
+      "maximum time to wait (in seconds) for the server to restart after a "
+      "restore operation before reporting an error; if zero, arangobackup will "
+      "not wait to check that the server restarts and will simply return the "
+      "result of the restore request",
+      new DoubleParameter(&_options.maxWaitForRestart));
+
+  options->addOption("--save-current",
+                     "whether to save the current state as a backup before "
+                     "restoring to another state",
+                     new BooleanParameter(&_options.saveCurrent));
+
+  /*
+    options->addSection(
+        "remote", "Options detailing a remote connection to use for operations");
+
+    options->addOption("--remote.credentials",
+                       "the credentials used for the remote endpoint (see manual "
+                       "for more details)",
+                       new StringParameter(&_options.credentials));
+
+    options->addOption("--remote.endpoint",
+                       "the remote endpoint (see manual for more details)",
+                       new StringParameter(&_options.endpoint));
+  */
+}
+
+void BackupFeature::validateOptions(std::shared_ptr<options::ProgramOptions> options) {
+  auto const& positionals = options->processingResult()._positionals;
+
+  if (1 == positionals.size()) {
+    _options.operation = positionals[0];
+  } else {
+    LOG_TOPIC(FATAL, Logger::BACKUP)
+        << "expected exactly one operation, got '"
+        << basics::StringUtils::join(positionals, ",") << "'";
```

Comment:
"," intentional or ", " (with space) really the desired separator?

Reply:
With space, good catch.